cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -73,6 +73,54 @@
 CLUSTER_NAME_AND_CUSTOM_DOMAIN_NAME_MAX_LENGTH = 255 - HOST_NAME_MAX_LENGTH - 1
 
 
+class FlexibleInstanceTypesValidator(Validator):
+    """Validator for Compute Resources that has multiple instance types."""
+
+    def _validate(
+        self,
+        queue_name: str,
+        compute_resource_name: str,
+        instance_types_info: Dict[str, InstanceTypeInfo],
+        disable_simultaneous_multithreading: bool,
+        efa_enabled: bool,
+        placement_group_enabled: bool,
+    ):
+        self.validate_cpu_requirements(compute_resource_name, instance_types_info, disable_simultaneous_multithreading)
+
+    def validate_size(self, items, size, failure_message, failure_level):
+        """Check if a list of items has a specific size and add a failure entry if it's exceeded."""
+        if len(items) > size:
+            self._add_failure(failure_message, failure_level)
+
+    def validate_cpu_requirements(
+        self,
+        compute_resource_name: str,
+        instance_types_info: Dict[str, InstanceTypeInfo],
+        disable_simultaneous_multithreading: bool,
+    ):
+        """Confirm CPU requirements for Flexible Instance Types.
+
+        Instance types should have the same number of CPUs or same number of Cores if Simultaneous Multithreading
+        is disabled.
+        """
+        if disable_simultaneous_multithreading:
+            self.validate_size(
+                {instance_type_info.cores_count() for instance_type_info in instance_types_info.values()},
```

Comment:
code it's not optimal since the check can fail as soon as there is a mismatch, without having to traverse all the list. But deviation isn't much if the list isn't big. Let's follow up on this